### ISSUE TITLE
VMD values to find executable for PRODY

### ISSUE THREAD

**Issue by vibhorag** [prody/ProDy#246]
Hello:
I was trying to use prody for the PCA calcuations.When VMD ask to specify the path for Prody and python.I supplied Python executable but prody execuatble is not recognised as in Python/Scripts folder there is prody but its not application.COuld you help to find the prody application  in python.Is it missing ?
Thanks
Vibhor

**Comment by cihankayacihan** [comment]
Prody executable is in ProDy/scripts. Try that prody.bat and please let us know if it will not work. Thanks.

**Comment by cmtshengu** [comment]
I have a customer with a similar issue. Here's her response to trying the prody.bat..."The prody.bat file you mention is the one I've tried initially that was giving the error that the file selected is not a prody executable.
It did not work"

Any other suggestions?

**Comment by cmtshengu** [comment]
Trying the prody.bat file again as mentioned in github gives the following error:


File "C:/Python27/Scripts/prody.bat", line 1
%~dp0\..\python %~dp0\prody %*
^
SyntaxError: invalid syntax
File "C:/Python27/Scripts/prody.bat", line 1
%~dp0\..\python %~dp0\prody %*
^
SyntaxError: invalid syntax
while executing
"exec $::NMWiz::pybin $::NMWiz::prody {*}$args"
(procedure "::NMWiz::prodySubmitPCAjob" line 61)
invoked from within
"::NMWiz::prodySubmitPCAjob"
(procedure "::NMWiz::prodySubmitJob" line 26)
invoked from within
"::NMWiz::prodySubmitJob"
invoked from within
".nmwizprody.submitFrame.prodySubmit invoke"
("uplevel" body line 1)
invoked from within
"uplevel #0 [list $w invoke]"
(procedure "tk::ButtonUp" line 24)
invoked from within
"tk::ButtonUp .nmwizprody.submitFrame.prodySubmit"
(command bound to event)

**Comment by ondopasa** [comment]
Hello!  I am having the same problem trying to use Prody though NMWiz in VMD. Could you solve the problem? Thanks in advance for your comments

**Comment by cmtshengu** [comment]
From https://github.com/prody/ProDy/issues/188#issuecomment-188308566

@cihankayacihan wrote

It is an issue with Windows version of Prody. It will be fixed soon.

**Comment by ondopasa** [comment]
Thank you for your answer @cihankayacihan. Fortunalely at last I could install it in Linux....

**Comment by cihankayacihan** [comment]
Hi. 

There are a couple issues related to NMWiz are addressed with last commits. Please see https://github.com/prody/NMWiz . There are two issues: 
1 -) Windows does not let you save pdb files in Program Files directory unless you have administrative privileges. Therefore, the current directory for prody interface in VMD changed to home directory. 
2-) Calling prody as python prody.bat in windows throws an error with new windows versions. Therefore, we throw away python part from the script. It is not required to specify python path but we keep it as a sanity check to avoid users trying to use prody from vmd without installing python.

**Comment by SamVas** [comment]
Hi all.
I am having the same issue on Mac. I am trying to use the prody interface in VMD through the Normal Mode Wizard, and I keep getting this message that I have to specify the prody path. However I am unable to locate the prody executable. Where is this file located on a Mac? My VMD version is 1.9.3 and I have the two nmwiz tcl files located here: /Applications/VMD\ 1.9.3.app/Contents/vmd/plugins/noarch/tcl/nmwiz1.2/
I appreciate your help.
Samantha

**Comment by gelusiji** [comment]
@cihankayacihan 
Hi cihankayacihan:
I run into the same problem as @vibhorag, then I downloaded the nmwiz.tcl updated file for Windows from the link you provided as https://github.com/prody/NMWiz
 and replaced the older one in my folder.
However, this time, when I launch Normal Mode Wizard, it failed, saying 
ERROR) Creation of window for 'nmwiz' failed (invalid command name "nmwiz_tk")
Could you please help me out of it?
I'm also using the Windows OS for your information.

Thanks a lot
Sean

**Comment by jamesmkrieger** [comment]
@SamVas 
You need to install ProDy separately. Please see http://prody.csb.pitt.edu/downloads/

@gelusiji
ok we'll look into it

**Comment by jamesmkrieger** [comment]
I have tried using the github NMWiz on my windows machine and Sean's problem doesn't happen so I guess this is fixed.

**Comment by jamesmkrieger** [comment]
I think the issue with Windows for the ProDy scripts is also fixed so I'm closing this issue

**Comment by andersfrederiksenQBL** [comment]
I have tried to replace the VMD folder with the git folder and installed ProDy seperately but I keep getting the error message

"C:\Users\MacOS>C:\Users\MacOS\DOCUME~1\ProDy-2.0\scripts\\..\python C:\Users\MacOS\DOCUME~1\ProDy-2.0\scripts\\prody anm --quiet -s all -o C:/Users/MacOS -p C:/Users/MacOS/AtCry_FADH_psf_anm -n 10 -c 15 -g 1 C:/Users/MacOS/AtCry_FADH_psf_anm.pdb 
'C:\Users\MacOS\DOCUME~1\ProDy-2.0\scripts\\..\python' is not recognized as an internal or external command,
operable program or batch file" 
when selecting the prody batch file in /PATH/ProDy-2.0/scripts folder

**Comment by jamesmkrieger** [comment]
You replaced the NMWiz that comes with VMD for the one from GitHub and installed ProDy separately and added the ProDy-2.0/scripts directly to your PATH, which is all correct. Therefore something else is wrong.

Do you also have python in your PATH? It looks like the problem is that it's trying to find python and can't. It seems to be looking just outside your ProDy-2.0/scripts directory, but that may not make sense seeing as you have your own ProDy installation outside the main Anaconda directory tree. I think you need to tell it separately where it is.

**Comment by andersfrederiksenQBL** [comment]
Yes, I have python in my PATH i.e. if I open a cmd and type python it opens a python shell
Furthermore I did not use Conda when installing ProDy but extracted it through the TAR file as their website tells i.e.
$ tar -xzf ProDy-x.y.z.tar.gz
$ cd ProDy-x.y.z
$ python setup.py build
$ python setup.py install
I hope this answers your question, if not you might have to dumb it down for me

**Comment by jamesmkrieger** [comment]
Ok, I see. So there is only the system python that you used to install ProDy. The commands you used should have created a new prody directory where all your other python modules go when you install them. I think you need to add the scripts directory from there to your path instead of the original one that you created with the tar -xzf command. That might then make it work.

**Comment by andersfrederiksenQBL** [comment]
Indeed a new prody folder was created, although for some reason the scripts folder has not been created in that. [here is a screenshot](https://imgur.com/a/YXJ29Lo).
Could it be I should unpack the ProDy folder in another folder?

**Comment by jamesmkrieger** [comment]
No, that's exactly what I said. You have a prody folder inside the ProDy 2.0 one in Documents but scripts isn't there. The scripts folder will be somewhere else where prody actually got installed. The best way to find it is probably to open a python terminal and type the following:
```
import prody
prody.__path__
```

See also https://stackoverflow.com/questions/122327/how-do-i-find-the-location-of-my-python-site-packages-directory for more information and ways of finding it.

**Comment by andersfrederiksenQBL** [comment]
I have now located the folder you ask for however there is still no scripts folder inside that prody folder

**Comment by jamesmkrieger** [comment]
What is that directory and what's in the directory above it?

**Comment by jamesmkrieger** [comment]
It might be a few up and then back down again.

**Comment by andersfrederiksenQBL** [comment]
[here are some screenshots of the folder and subfolder and another Prody folder](https://imgur.com/a/BQNr3VY)

**Comment by jamesmkrieger** [comment]
Can you try looking above Lib in the Python39 folder? I have a Scripts folder in a place equivalent to that on my computer

**Comment by andersfrederiksenQBL** [comment]
Ah, that's where it was hiding, now it works, thank you for the help

**Comment by jamesmkrieger** [comment]
You're very welcome. 

Issue closed.
